mmcv/cnn/bricks/transformer.py

```diff
@@ -0,0 +1,599 @@
+import torch
+import torch.nn as nn
+
+from ..utils import xavier_init
+from .activation import build_activation_layer
+from .norm import build_norm_layer
+from .wrappers import Linear
+
+
+class MultiheadAttention(nn.Module):
+    """A warpper for torch.nn.MultiheadAttention.
+
+    This module implements MultiheadAttention with residual connection.
+
+    Args:
+        feat_channels (int): The embedding dimension.
+        num_heads (int): Parallel attention heads. Same as
+            `nn.MultiheadAttention`.
+        dropout (float): A Dropout layer on attn_output_weights. Default 0.0.
+    """
+
+    def __init__(self, feat_channels, num_heads, dropout=0.0):
+        super(MultiheadAttention, self).__init__()
+        assert feat_channels % num_heads == 0, 'feat_channels must be ' \
+            f'divisible by num_heads. got {feat_channels} and {num_heads}.'
+        self.feat_channels = feat_channels
+        self.num_heads = num_heads
+        self.dropout = dropout
+        self.attn = nn.MultiheadAttention(feat_channels, num_heads, dropout)
+        self.dropout = nn.Dropout(dropout)
+
+    def forward(self,
+                x,
+                key=None,
+                value=None,
+                residual=None,
+                query_pos=None,
+                key_pos=None,
+                attn_mask=None,
+                key_padding_mask=None):
+        """Forward function for MultiheadAttention."""
```

Comment:
This function is complicated and has many arguments. We may use a standard docstring.